MeetupPlatformApi/Controllers/AuthenticationController.cs

```diff
@@ -44,33 +44,99 @@ public async Task<IActionResult> RegisterUser([FromBody] UserRegistrationDto reg
         }
 
         var user = mapper.Map<UserEntity>(registrationDto);
-
         user.Password = BCrypt.HashPassword(user.Password);
+        await context.Users.AddAsync(user);
 
-        context.Users.Add(user);
+        var tokenPair = authenticationManager.IssueTokenPair(user);
+        var refreshToken = new RefreshTokenEntity()
+        {
+            Id = authenticationManager.GetNameIdentifier(tokenPair.RefreshToken),
+            Expires = authenticationManager.GetExpires(tokenPair.RefreshToken),
+            UserId = user.Id
+        };
+        await context.RefreshTokens.AddAsync(refreshToken);
         await context.SaveChangesAsync();
 
-        var accessToken = authenticationManager.IssueAccessToken(user);
         var userInfoDto = mapper.Map<UserOutputDto>(user);
         var outputDto = new UserRegistrationResultDto
         {
             UserInfo = userInfoDto,
-            AccessToken = accessToken
+            TokenPair = new TokenPairDto
```

Comment:
Consider using AutoMapper here.
I'm not strongly advocating for doing so, but we also can't completely ignore this option.